### PR TITLE
docs(angular): Add support for Angular version 15+

### PR DESCRIPTION
**Pull request by rifont**
**Summary**
* Update Angular client and quickstart docs to specify support for version 15+, per https://github.com/novuhq/novu/pull/4518
* Formatting fixes on Angular quickstart
  * Remove extra lines
  * Add code-blocks for install CLI commands
* Adds a snippet for supported channels with some visual tidy-up

**Screenshots**
_Angular Client_
image

_Supported Channels Snippet_
image

### CODE REVIEW

**Comment by rifont** [comment]
We'll need to hold off on merging this until we release support for Angular 15+ in v0.20, as part of https://github.com/novuhq/novu/pull/4518

File: _snippets/supported-channels.mdx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+| Channel<br/><br/> | Content Style<br/><br/> | Custom Variables<br/>`{{handlebars}}` format |
+| --- | --- | :-: |
+| **Email** | HTML | ✅ |
+|  | Visual Editor | ✅ |
+| **SMS** | Text | ✅ |
+| **Chat** | Text | ✅ |
+| **In-App** | Text | ✅ |
+| **Push** | Text | ✅ |
```
Comment (rifont) [inline _snippets/supported-channels.mdx:8]: This has 20+ occurrences across the docs, and is likely to change as we support more channels, so I've created a snippet.

File: notification-center/client/angular.mdx [hunk 2]
```diff
@@ -8,11 +8,19 @@ The `@novu/notification-center-angular` package provides an Angular component
...
+```bash pnpm
+pnpm i @novu/notification-center-angular
+```
+```bash npm
+npm i @novu/notification-center-angular
+```
+```bash yarn
+yarn add @novu/notification-center-angular
+```
+</CodeGroup>
```
Comment (rifont) [inline notification-center/client/angular.mdx:23]: `CodeGroup` for a nicer experience to install. Defaults to `pnpm`

File: notification-center/client/angular.mdx [hunk 3]
```diff
@@ -63,9 +71,7 @@ Component HTML template:
 The `notification-center-component` accepts the same set of props as the [web component](/notification-center/client/web-component).
 
 <Tip>
-  May need to add `"allowSyntheticDefaultImports": true` in tsconfig.json
-  and `@types/react` as dev dependency for the angular component to work
-  properly
+  You may need to add `"allowSyntheticDefaultImports": true` in tsconfig.json for the web component's React imports to work correctly.
```
Comment (rifont) [inline notification-center/client/angular.mdx:74]: No need to install `@types/react` anymore per https://github.com/novuhq/novu/pull/4518.

File: quickstarts/angular.mdx [hunk 4]
```diff
@@ -52,44 +49,44 @@ The Novu Angular package provides a Angular component wrapper over the web compo
...
   subscriberId: '',
   applicationIdentifier: '',
 };
-
 ```
 
 Copy and paste the same code into `my-app/src/environments/environment.development.ts`
 
-These variables are needed for the `GET` request our notification center will make to Novu’s API to actually push notifications into the feed.
+These variables are needed for the `GET` request our notification center will make to Novu’s API to push notifications into the feed.
```
Comment (rifont) [inline quickstarts/angular.mdx:89]: Had some non-standard space characters here previously.

File: quickstarts/angular.mdx [hunk 5]
```diff
@@ -183,63 +179,20 @@ export class AppComponent {
...
+Now head to the `my-app/tsconfig.json` file, we’re going to add `"allowSyntheticDefaultImports": true` to the `compilerOptions` array.
 
-Open your terminal and navigate to the app root directory and type the following:
-
-```
-npm i [@types/react](https://github.com/types/react)
-
-or
-
-yarn add [@types/react](https://github.com/types/react)
```
Comment (rifont) [inline quickstarts/angular.mdx:202]: As above, no longer required.

File: quickstarts/angular.mdx [hunk 6]
```diff
@@ -183,63 +179,20 @@ export class AppComponent {
...
-    "module": "ES2022",
-    "useDefineForClassFields": false,
-    "lib": ["ES2022", "dom"]
-  },
-  "angularCompilerOptions": {
-    "enableI18nLegacyMessageIdFormat": false,
-    "strictInjectionParameters": true,
-    "strictInputAccessModifiers": true,
-    "strictTemplates": true
++   "allowSyntheticDefaultImports": true,
```
Comment (rifont) [inline quickstarts/angular.mdx:193]: Simplifying this down into a diff, as some users may have a different looking `tsconfig` they are integrating with.

File: quickstarts/angular.mdx [hunk 7]
```diff
@@ -253,15 +206,14 @@ We will add our notification center to the .toolbar div.
 Paste the following into your `app.component.html` file:
 
 ```jsx
-<div id="bell-icon">
+  <div id="bell-icon">
```
Comment (rifont) [inline quickstarts/angular.mdx:209]: Out of the box alignment into the `.html` file.

**Review by rifont — COMMENTED** [review]
I have made a note on Linear to merge this PR when `v0.21.0` is released.

It is ready for review.

**Review by scopsy — APPROVED** [review]

**Pull request merged**